src/ml/surrogate.hpp

```diff
@@ -150,21 +176,36 @@ class SurrogateModel
                         size_t num_in,
                         size_t num_out,
                         const TypeInValue** inputs,
-                        TypeInValue** outputs)
+                        TypeInValue** outputs,
+                        TypeInValue** outputs_stdev)
   {
     //torch::NoGradGuard no_grad;
     c10::InferenceMode guard(true);
     auto input = arrayToTensor(num_elements, num_in, inputs);
     input.set_requires_grad(false);
-    at::Tensor output = module.forward({input}).toTensor().detach();
+    if (_isDeltaUq) {
+      assert(outputs_stdev && "Expected non-null outputs_stdev");
+      // The deltauq surrogate returns a tuple of (outputs, outputs_stdev)
+      auto output_tuple = module.forward({input}).toTuple();
+      at::Tensor output_tensor = output_tuple->elements()[0].toTensor().detach();
```

Comment:
Do you want to call this output_mean_tensor?

Reply:
If you insist

Reply:
Done